packages/nx/src/utils/nx-plugin.ts

```diff
@@ -15,39 +19,87 @@ import {
 } from '../plugins/js/utils/register';
 import {
   ProjectConfiguration,
-  TargetConfiguration,
+  ProjectsConfigurations,
 } from '../config/workspace-json-project-json';
 import { logger } from './logger';
 import {
   createProjectRootMappingsFromProjectConfigurations,
   findProjectForPath,
 } from '../project-graph/utils/find-project-for-path';
 import { normalizePath } from './path';
-import { join } from 'path';
+import { dirname, join } from 'path';
 import { getNxRequirePaths } from './installation-directory';
 import { readTsConfig } from '../plugins/js/utils/typescript';
+import { NxJsonConfiguration } from '../config/nx-json';
 
 import type * as ts from 'typescript';
 import { retrieveProjectConfigurationsWithoutPluginInference } from '../project-graph/utils/retrieve-workspace-files';
+import { NxPluginV1 } from './nx-plugin.deprecated';
+import { ProjectGraphDependencyWithFile } from '../project-graph/project-graph-builder';
+import { combineGlobPatterns } from './globs';
+
+export type ProjectConfigurationsConstructor = [
+  projectFilePattern: string,
+  constructorMethod: (
+    projectConfigurationFile: string,
+    context: {
+      projectsConfigurations: Record<string, ProjectConfiguration>;
+      nxJsonConfiguration: NxJsonConfiguration;
+      workspaceRoot: string;
+    }
+  ) => {
+    projectNodes?: Record<string, ProjectConfiguration>;
```

Comment:
The property is projectNodes but the type is project configurations. The name should align?